clang-tools-extra/clang-tidy/performance/UnnecessaryCopyInitialization.h

```diff
@@ -32,14 +32,34 @@ class UnnecessaryCopyInitialization : public ClangTidyCheck {
   void check(const ast_matchers::MatchFinder::MatchResult &Result) override;
   void storeOptions(ClangTidyOptions::OptionMap &Opts) override;
 
+protected:
+  // A helper to manipulate the state common to
+  // `CopyFromMethodReturn` and `CopyFromLocalVar`.
+  struct CheckContext {
+    CheckContext(const ast_matchers::MatchFinder::MatchResult &Result);
+    const VarDecl &Var;
+    const Stmt &BlockStmt;
+    const DeclStmt &VarDeclStmt;
+    clang::ASTContext &ASTCtx;
+    const bool IssueFix;
+    const bool IsVarUnused;
+    const bool IsVarOnlyUsedAsConst;
+  };
+
+  // Create diagnostics. These are virtual so that derived classes can change
+  // behaviour.
+  virtual void diagnoseCopyFromMethodReturn(const CheckContext &Ctx,
```

Comment:
do not mix references with pointers, stick to one, specially in functions that does similar thing

Reply:
`ObjectArg` is a pointer because it can be null (unlike other variables, which are always non-null. Anyway, this is no longer provided as pre the previous comment.